pandas/core/dtypes/generic.py

```diff
@@ -23,24 +37,46 @@ def _check(cls, inst) -> bool:
     return meta(name, tuple(), dct)
 
 
-ABCIndex = create_pandas_abc_type("ABCIndex", "_typ", ("index",))
-ABCInt64Index = create_pandas_abc_type("ABCInt64Index", "_typ", ("int64index",))
-ABCUInt64Index = create_pandas_abc_type("ABCUInt64Index", "_typ", ("uint64index",))
-ABCRangeIndex = create_pandas_abc_type("ABCRangeIndex", "_typ", ("rangeindex",))
-ABCFloat64Index = create_pandas_abc_type("ABCFloat64Index", "_typ", ("float64index",))
-ABCMultiIndex = create_pandas_abc_type("ABCMultiIndex", "_typ", ("multiindex",))
-ABCDatetimeIndex = create_pandas_abc_type(
-    "ABCDatetimeIndex", "_typ", ("datetimeindex",)
-)
-ABCTimedeltaIndex = create_pandas_abc_type(
-    "ABCTimedeltaIndex", "_typ", ("timedeltaindex",)
-)
-ABCPeriodIndex = create_pandas_abc_type("ABCPeriodIndex", "_typ", ("periodindex",))
-ABCCategoricalIndex = create_pandas_abc_type(
-    "ABCCategoricalIndex", "_typ", ("categoricalindex",)
-)
-ABCIntervalIndex = create_pandas_abc_type(
-    "ABCIntervalIndex", "_typ", ("intervalindex",)
+ABCIndex = cast("Type[Index]", create_pandas_abc_type("ABCIndex", "_typ", ("index",)))
```

Comment:
neat idea

Reply:
actually, ABCIndex might be weird one, since it is only Index but not subclasses

Reply:
Just to be sure - this is @simonjayhawkins's idea. You may remember mine, it was far worse :D

Regarding the subclass issue - this won't change the behavior of isinstance checks, only what mypy is able to infer from them.

Reply:
sounds good, thanks